pkg/util/downloader/downloader.go

```diff
@@ -38,5 +32,10 @@ func Download(url, filename, targetDir string) (int64, error) {
 		return 0, err
 	}
 
+	resp, err := http.Get(url)
```

Comment:
Very good adjustment. No need to download the file first if we can't even get the filename from the URL.